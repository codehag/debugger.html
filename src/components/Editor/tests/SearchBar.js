import { createFactory } from "react";
import { shallow } from "enzyme";
import SearchBar from "../SearchBar";

const SearchBarComponent = createFactory(SearchBar.WrappedComponent);

function generateDefaults() {
  return {
    query: "",
    searchOn: true,
    searchResults: [],
    selectedResultIndex: 0
  };
}

function render(overrides = {}) {
  const defaults = generateDefaults();
  const props = { ...defaults, ...overrides };
  const component = shallow(new SearchBarComponent(props));
  return { component, props };
}

describe("SearchBar", () => {
  it("should render", () => {
    const { component } = render();
    expect(component).toMatchSnapshot();
  });

  it("should have a result list with search results from the props", () => {
    const searchResults = [1, 2, 3];
    const { component } = render({ searchResults });
    const resultList = component.find("ResultList");
    expect(resultList.prop("items")).toBe(searchResults);
  });
});
