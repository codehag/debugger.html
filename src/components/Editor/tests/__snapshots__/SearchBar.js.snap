// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should render 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="search-bar"
>
    <SearchInput
        count={undefined}
        handleClose={[Function]}
        handleNext={[Function]}
        handlePrev={[Function]}
        onChange={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        placeholder="Search in file…"
        query=""
        size=""
        summaryMsg=""
    />
    <div
        className="search-bottom-bar"
    >
        <section
            className="search-type-toggles"
        >
            <h1
                className="search-toggle-title"
            >
                Search for:
            </h1>
            <button
                className="search-type-btn"
                onClick={[Function]}
            >
                functions
            </button>
            <button
                className="search-type-btn"
                onClick={[Function]}
            >
                variables
            </button>
        </section>
        <div
            className="search-modifiers"
        >
            <button
                className="regex-match-btn"
                onClick={[Function]}
                title="Regex"
            >
                <InlineSVG
                    className="regex-match"
                    element="i"
                    raw={false}
                    src="<svg></svg>"
                />
            </button>
            <button
                className="case-sensitive-btn"
                onClick={[Function]}
                title="Case sensitive"
            >
                <InlineSVG
                    className="case-match"
                    element="i"
                    raw={false}
                    src="<svg></svg>"
                />
            </button>
            <button
                className="whole-word-btn"
                onClick={[Function]}
                title="Whole word"
            >
                <InlineSVG
                    className="whole-word-match"
                    element="i"
                    raw={false}
                    src="<svg></svg>"
                />
            </button>
        </div>
    </div>
</div>,
  "nodes": Array [
    <div
      className="search-bar"
>
      <SearchInput
            count={undefined}
            handleClose={[Function]}
            handleNext={[Function]}
            handlePrev={[Function]}
            onChange={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            placeholder="Search in file…"
            query=""
            size=""
            summaryMsg=""
      />
      <div
            className="search-bottom-bar"
      >
            <section
                  className="search-type-toggles"
            >
                  <h1
                        className="search-toggle-title"
                  >
                        Search for:
                  </h1>
                  <button
                        className="search-type-btn"
                        onClick={[Function]}
                  >
                        functions
                  </button>
                  <button
                        className="search-type-btn"
                        onClick={[Function]}
                  >
                        variables
                  </button>
            </section>
            <div
                  className="search-modifiers"
            >
                  <button
                        className="regex-match-btn"
                        onClick={[Function]}
                        title="Regex"
                  >
                        <InlineSVG
                              className="regex-match"
                              element="i"
                              raw={false}
                              src="<svg></svg>"
                        />
                  </button>
                  <button
                        className="case-sensitive-btn"
                        onClick={[Function]}
                        title="Case sensitive"
                  >
                        <InlineSVG
                              className="case-match"
                              element="i"
                              raw={false}
                              src="<svg></svg>"
                        />
                  </button>
                  <button
                        className="whole-word-btn"
                        onClick={[Function]}
                        title="Whole word"
                  >
                        <InlineSVG
                              className="whole-word-match"
                              element="i"
                              raw={false}
                              src="<svg></svg>"
                        />
                  </button>
            </div>
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <SearchBar
        query=""
        searchOn={true}
        searchResults={Array []}
        selectedResultIndex={0}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": SearchBar {
        "_reactInternalInstance": [Circular],
        "buildPlaceHolder": [Function],
        "buildSummaryMsg": [Function],
        "clearSearch": [Function],
        "closeSearch": [Function],
        "context": Object {
          "shortcuts": undefined,
        },
        "doSearch": [Function],
        "onChange": [Function],
        "onEscape": [Function],
        "onKeyDown": [Function],
        "onKeyUp": [Function],
        "onSelectResultItem": [Function],
        "props": Object {
          "query": "",
          "searchOn": true,
          "searchResults": Array [],
          "selectedResultIndex": 0,
        },
        "refs": Object {},
        "renderBottomBar": [Function],
        "renderResults": [Function],
        "renderSearchModifiers": [Function],
        "renderSearchTypeToggle": [Function],
        "searchContents": [Function],
        "searchInput": [Function],
        "selectResultItem": [Function],
        "selectSearchInput": [Function],
        "setSearchValue": [Function],
        "state": Object {
          "count": 0,
          "index": -1,
          "selectedResultIndex": 0,
          "symbolSearchResults": Array [],
        },
        "toggleSearch": [Function],
        "toggleSymbolSearch": [Function],
        "traverseCodeResults": [Function],
        "traverseResults": [Function],
        "traverseSymbolResults": [Function],
        "updateSymbolSearchResults": [Function],
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": Object {
        "_currentElement": <div
          className="search-bar"
>
          <SearchInput
                    count={undefined}
                    handleClose={[Function]}
                    handleNext={[Function]}
                    handlePrev={[Function]}
                    onChange={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    placeholder="Search in file…"
                    query=""
                    size=""
                    summaryMsg=""
          />
          <div
                    className="search-bottom-bar"
          >
                    <section
                              className="search-type-toggles"
                    >
                              <h1
                                        className="search-toggle-title"
                              >
                                        Search for:
                              </h1>
                              <button
                                        className="search-type-btn"
                                        onClick={[Function]}
                              >
                                        functions
                              </button>
                              <button
                                        className="search-type-btn"
                                        onClick={[Function]}
                              >
                                        variables
                              </button>
                    </section>
                    <div
                              className="search-modifiers"
                    >
                              <button
                                        className="regex-match-btn"
                                        onClick={[Function]}
                                        title="Regex"
                              >
                                        <InlineSVG
                                                  className="regex-match"
                                                  element="i"
                                                  raw={false}
                                                  src="<svg></svg>"
                                        />
                              </button>
                              <button
                                        className="case-sensitive-btn"
                                        onClick={[Function]}
                                        title="Case sensitive"
                              >
                                        <InlineSVG
                                                  className="case-match"
                                                  element="i"
                                                  raw={false}
                                                  src="<svg></svg>"
                                        />
                              </button>
                              <button
                                        className="whole-word-btn"
                                        onClick={[Function]}
                                        title="Whole word"
                              >
                                        <InlineSVG
                                                  className="whole-word-match"
                                                  element="i"
                                                  raw={false}
                                                  src="<svg></svg>"
                                        />
                              </button>
                    </div>
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="search-bar"
>
          <SearchInput
                    count={undefined}
                    handleClose={[Function]}
                    handleNext={[Function]}
                    handlePrev={[Function]}
                    onChange={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    placeholder="Search in file…"
                    query=""
                    size=""
                    summaryMsg=""
          />
          <div
                    className="search-bottom-bar"
          >
                    <section
                              className="search-type-toggles"
                    >
                              <h1
                                        className="search-toggle-title"
                              >
                                        Search for:
                              </h1>
                              <button
                                        className="search-type-btn"
                                        onClick={[Function]}
                              >
                                        functions
                              </button>
                              <button
                                        className="search-type-btn"
                                        onClick={[Function]}
                              >
                                        variables
                              </button>
                    </section>
                    <div
                              className="search-modifiers"
                    >
                              <button
                                        className="regex-match-btn"
                                        onClick={[Function]}
                                        title="Regex"
                              >
                                        <InlineSVG
                                                  className="regex-match"
                                                  element="i"
                                                  raw={false}
                                                  src="<svg></svg>"
                                        />
                              </button>
                              <button
                                        className="case-sensitive-btn"
                                        onClick={[Function]}
                                        title="Case sensitive"
                              >
                                        <InlineSVG
                                                  className="case-match"
                                                  element="i"
                                                  raw={false}
                                                  src="<svg></svg>"
                                        />
                              </button>
                              <button
                                        className="whole-word-btn"
                                        onClick={[Function]}
                                        title="Whole word"
                              >
                                        <InlineSVG
                                                  className="whole-word-match"
                                                  element="i"
                                                  raw={false}
                                                  src="<svg></svg>"
                                        />
                              </button>
                    </div>
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <SearchBar
    query=""
    searchOn={true}
    searchResults={Array []}
    selectedResultIndex={0}
/>,
}
`;
